src/bessel.jl

```diff
@@ -590,6 +590,27 @@ function bessely(n::Integer, x::BigFloat)
     return z
 end
 
+softinvsqrt(x)          = sqrt(x)/(abs(x) + (eps ∘ float      )(x))
+softinvsqrt(z::Complex) = sqrt(z)/(abs(z) + (eps ∘ float ∘ abs)(z))
+
+"""
+    sphericalbesselj(nu, x)
+
+Spherical bessel function of the first kind at order \$nu\$, \$j_\\nu(x)\$. This is the non-singular
+solution to the radial part of the Helmholz equation in spherical coordinates.
+"""
+sphericalbesselj(nu, x) = √(π/2)*softinvsqrt(x)*besselj(nu+1/2, x)
```

Comment:
```suggestion
sphericalbesselj(nu, x) = √(π/2) * softinvsqrt(x) * besselj(nu + 1//2, x)
```
Using a rational there avoids immediate promotion to `Float64`.

Reply:
Isn't `π/2` also Float64? So perhaps `sqrt(T(pi)/2)`.

Reply:
Good catch, @mcabbott. That is indeed better.